backend/clubs/views.py

```diff
@@ -2878,33 +2878,64 @@ def get(self, request):
                                     type: integer
         ---
         """
-        id = request.query_params.get("event")
-        if not id:
-            return Response({"count": 0})
-
-        try:
-            conn = get_redis_connection()
-        except NotImplementedError:
-            return Response({"count": 0})
-        key = f"zoom:meeting:live:{id}"
-        ans = conn.get(key)
-        if ans is None:
-            ans = 0
-        else:
-            ans = int(ans)
-        return Response({"count": ans})
+        meeting_id = request.query_params.get("event")
+        return Response({"count": ZoomMeetingVisit.objects.filter(meeting_id=meeting_id).count()})
 
     def post(self, request):
         action = request.data.get("event")
-        meeting_id = request.data.get("payload", {}).get("object", {}).get("id", None)
-        if meeting_id is not None:
-            conn = get_redis_connection()
-            key = f"zoom:meeting:live:{meeting_id}"
-            if action == "meeting.participant_joined":
-                conn.incr(key)
-            elif action == "meeting.participant_left":
-                conn.decr(key)
-            conn.expire(key, datetime.timedelta(hours=8))
+        if action == "meeting.participant_joined":
+            email = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("email", None)
+            )
+            person = get_user_model().objects.filter(email=email)
```

Comment:
I'm not positive that this is the best way to get the user (filtering the user model) but I couldn't think of any other way to get the user from our DB using the information provided by the webhook API

Reply:
This won't actually work, as the Penn emails are `pennkey@seas.upenn.edu`, etc while the Zoom ones are `pennkey@upenn.edu`. You instead want to parse the username from the email and do matching on the username.

Reply:
Also handle the edge case where there is no username match. This could happen if someone gets the link from their friend, never logged into Penn Clubs before, and joins the meeting.

In this case, you could set this field to null and make the person foreign key nullable.